src/app/zap-templates/templates/app/helper.js

```diff
@@ -223,8 +252,33 @@ function chip_endpoint_cluster_list()
     } else {
       mask = c.mask.map((m) => `ZAP_CLUSTER_MASK(${m.toUpperCase()})`).join(' | ')
     }
-    ret = ret.concat(`  { ${c.clusterId}, ZAP_ATTRIBUTE_INDEX(${c.attributeIndex}), ${c.attributeCount}, ${c.attributeSize}, ${
-        mask}, ${functionArray} }, /* ${c.comment} */ \\\n`)
+
+    clientGeneratedCommands = [];
+    serverGeneratedCommands = [];
+    c.commands.forEach((cmd) => {
+      console.log(cmd);
+      if (cmd.mask.includes('incoming_server')) {
+        clientGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+      if (cmd.mask.includes('incoming_client')) {
+        serverGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+    });
+    clientGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
+    serverGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
```

Comment:
Why do we need to build these lists?  I would be tempted to do something like:
```suggestion
    let clientGeneratedCommandCount = 0;
    let serverGeneratedCommandCount = 0;
    c.commands.forEach((cmd) => {
      if (cmd.mask.includes('incoming_server')) {
        ++clientGeneratedCommandCount;
      }
      if (cmd.mask.includes('incoming_client')) {
        ++serverGeneratedCommandCount;
      }
    });
    let clientGeneratedCommandValue;
    if (clientGeneratedCommandCount != 0) {
        // Add 1 for the list terminator
        ++clientGeneratedCommandCount;
        clientGeneratedCommandValue = `ZAP_GENERATED_COMMANDS_INDEX(${totalCommands})`;
    } else {
        clientGeneratedCommandValue = "nullptr";
    }
    totalCommands += clientGeneratedCommandCount;
    
    // Same thing but for serverGeneratedCommandCount
```
and this way we avoid various 0xFFFFFFFF that are just marking empty lists.  The code that reads these values already does a null-check on the value it gets from the struct, so we can use null to represent empty list.

Reply:
Updated, we still need a 0xffffffff for end of list here.

Reply:
Only if the list is nonempty, because nothing points to empty lists.